### ISSUE TITLE
[ISSUE]LDAP Group Membership Authentication issue

### ISSUE THREAD

**Issue by hendrikb101** [WillyXJ/facileManager#479]
fM Version : 3.5.6
{fmDNS} Version : 3.4.1

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/WillyXJ/facileManager/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/WillyXJ/facileManager/issues?q=is%3Aissue+is%3Aclosed) and [opened issues](https://github.com/WillyXJ/facileManager/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

---

**(ISSUE) Expected Behavior:**
I am trying to authenticate FMManager to my LDAP Server with group membership but with no luck. (Same as issue #392 in 2017)

**(ISSUE) Actual Behavior:**

(FreeIPA) OpenLDAP server log :
BIND dn="uid=testuser,cn=users,cn=accounts,dc=domain,dc=co,dc=za" method=128 version=3
RESULT err=0 tag=97 nentries=0 etime=0.0002386482 dn="uid=testuser,cn=users,cn=accounts,dc=domain,dc=co,dc=za"
CMP dn="cn=fmmanager,cn=groups,cn=compat,dc=domain,dc=co,dc=za" attr="memberUid"
RESULT err=53 tag=111 nentries=0 etime=0.0000173325
UNBIND

Without group membership it's working fine, but I really need group membership enabled.
It's as if it's not calling the username after "memberUid"

I took the DN directly from the LDAP tree structure. Something I'm doing wrong?

Thanks in advance
**(ISSUE) Steps to reproduce:**

**Comment by WillyXJ** [comment]
What software versions are you using (php, ldap, etc.)?

**Comment by WillyXJ** [comment]
Are you specifying a valid group attribute?  That field is used to compare the user with the group.

```
$ldap_group_response = @ldap_compare($ldap_connect, $ldap_group_dn, $ldap_group_attribute, $username);
```

**Comment by hendrikb101** [comment]
Hi

I am using php72w and LDAP389 (build into FreeIPA, redhat application)

This is what my tree looks like.

# fmfirewall, groups, compat, domain.co
dn: cn=fmfirewall,cn=groups,cn=compat,dc=domain,dc=co
objectClass: posixGroup
objectClass: ipaOverrideTarget
objectClass: top
gidNumber: 228400007
memberUid: hennieb
ipaAnchorUUID:: OklQQTpiY3h3cy5jby56YTo4ZWFlMDVmNC02MzZhLTExZWEtYTE4Ny0wMDBjMj
kzYTNmNGM=
cn: fmfirewall

# fmfirewall, groups, accounts, domain.co
dn: cn=fmfirewall,cn=groups,cn=accounts,dc=domain,dc=co
objectClass: top
objectClass: groupofnames
objectClass: nestedgroup
objectClass: ipausergroup
objectClass: ipaobject
objectClass: posixgroup
cn: fmfirewall
description: Access Group for FacileManager Firewall application
ipaUniqueID: 8eae05f4-636a-11ea-a187-000c293a3f4c
gidNumber: 228400007

[cid:image002.png@01D5F78D.B2172B90]
When I use this config, this is the output on the LDAP server.

[11/Mar/2020:09:45:04.052681180 +0200] conn=87 fd=111 slot=111 connection from 10.1.98.22 to 10.1.98.25
[11/Mar/2020:09:45:04.053005669 +0200] conn=87 op=0 BIND dn="uid=hennieb,cn=users,cn=accounts,dc=domain,dc=co" method=128 version=3
[11/Mar/2020:09:45:04.053805869 +0200] conn=87 op=0 RESULT err=0 tag=97 nentries=0 etime=0.0001052608 dn="uid=hennieb,cn=users,cn=accounts,dc=domain,dc=co"
[11/Mar/2020:09:45:04.054243567 +0200] conn=87 op=1 CMP dn="memberOf=cn=fmfirewall,cn=groups,cn=accounts,dc=domain,dc=co" attr="memberUid"
[11/Mar/2020:09:45:04.054540586 +0200] conn=87 op=1 RESULT err=0 tag=111 nentries=0 etime=0.0000377154
[11/Mar/2020:09:45:04.054777295 +0200] conn=87 op=2 UNBIND
[11/Mar/2020:09:45:04.054797308 +0200] conn=87 op=2 fd=111 closed - U1
[11/Mar/2020:09:45:08.255995478 +0200] conn=86 op=7 UNBIND
[11/Mar/2020:09:45:08.256027424 +0200] conn=86 op=7 fd=108 closed - U1

Would really appreciate your help with this.

Thanks.

Hennie.

**Comment by WillyXJ** [comment]
I think all you are missing is to add “memberOf” in the “Group Attributes” in the fM settings.

**Comment by WillyXJ** [comment]
@hendrikb101  - Did this get resolved?

**Comment by hendrikb101** [comment]
Unfortunately not yet, it’s not passing the current attributes to the LDAP server, I will have a look at it again this week.

Hennie.

**Comment by WillyXJ** [comment]
@hendrikb101 - Are you able to confirm the fix provided in #533 fixes the authentication issue you've been experiencing?
